Client/TabManagement/Tab.swift

```diff
@@ -951,7 +951,7 @@ class TabWebView: WKWebView, MenuHelperInterface {
     var accessoryPreviousAction: (() -> Void)?
     var accessoryDoneAction: (() -> Void)?
     var accessoryCreditCardAction: (() -> Void)?
-    var accessoryView: AccessoryViewProvider? = AccessoryViewProvider()
+    var accessoryView: AccessoryViewProvider?
 
     override var inputAccessoryView: UIView? {
         guard let accessoryView = accessoryView else { return nil }
```

Comment:
Step2: You aren't initializing properly and thus my guess is the reason you have to do all the dance of reloading with the timer.

Once you fix the `accessoryView` initialization and do it inside `override init(frame: CGRect, configuration: WKWebViewConfiguration) it should work.

Reply:
Also, for override with correct initialization you will only need to `return accessoryView`

Reply:
In the current state the initialization is done through keyboardHelperDelegate's keyboardWillShowWithState. 

And changing to show the correct accessory view is only the first problem. Reseting the accessory view back to the standard one is the other. For that reason I still think having option 1 [from the comment](https://mozilla-hub.atlassian.net/browse/FXIOS-6297?focusedCommentId=678984) would be useful.

Reply:
Have you tried these changes I suggested, could you push so I can see what you have tried :)

Reply:
Sure, I'll push. Just want to say the specific case the delay is addressing:

- Go to a checkout page and have the keyboard NOT on screen.
- Tap a field that isn't CC input.
- Dismiss the keyboard.
- Tap on a CC input field
- Observe that the Use saved card button doesn't appear.

I'm still against having this delay present. I still think option 1 would help us address this particular case - we could reset the accessory view in there.

Reply:
@nishant2718 can you upload a video showing it isn't working? I see it working fine 🤷🏼‍♂️

Reply:
FYI `creditCardHelper.foundFieldValues = { [weak self] fieldValues in` is what governs whether you are in credit card field or not, I believe I mentioned that before. Take that closure as your reference point and update the view.

Reply:
Merging this in as we discussed offline that other improvements aren't part of this PR. It is working as intended and any improvements related to JS will follow-through in other PR